pkg/kptfile/kptfileutil/util_test.go

```diff
@@ -151,6 +153,342 @@ upstreamBadField:
 	assert.Equal(t, kptfilev1alpha2.KptFile{}, f)
 }
 
-func TestMerge(t *testing.T) {
+func TestUpdateKptfile(t *testing.T) {
```

Comment:
Upstream has a function with apply-setters image and default values for setters, local changes the value of setters, upstream adds new setter to the list, update would make setter values on local back to default ? This is not acceptable.

I would suggest having real world examples with known function images.

Reply:
After discussions, we chose this as the temporary solution until we can implement a solution where we can let users manually resolve conflicts. The challenge here is that even with the `Name` field, there isn't a way to merge lists so the order of elements in the resulting list is obvious to users. Everyone agrees that we should try to find a better way.

Reply:
@phanimarupaka since we don't have a good merge strategy that can handle common cases well automatically, we are sort of forced to go with this solution temporarily.